### PR TITLE
bug/go-2380 > Incorrect open balance at St.Jude Kenya instance 

### PR DESCRIPTION
**Pull request by kwburnett**
This is to continue on #36 but with further work.

### CODE REVIEW

**Comment by kwburnett** [comment]
I know you've already reviewed these code changes up to a certain point - would you mind reviewing the new additions?

File: idemp-base-test/src/org/bandahealth/idempiere/base/test/modelevent/InvoiceModelEventTest.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		ChuBoeCreateEntity.createInvoice(valueObject);
+		commitEx();
+
+		ordersInvoices = valueObject.getOrder().getInvoices();
+		assertEquals(1, ordersInvoices.length, "Invoice was created for order");
+		assertTrue(Arrays.stream(ordersInvoices[0].getLines(true)).noneMatch(
+				invoiceLine -> invoiceLine.getC_Charge_ID() > 0 &&
+						invoiceLine.getC_Charge_ID() == valueObject.getCharge().get_ID()), "Invoice doesn't have charge");
+	}
+}
```
Comment (kwburnett) [inline idemp-base-test/src/org/bandahealth/idempiere/base/test/modelevent/InvoiceModelEventTest.java]:
```suggestion
}

```

File: idemp-base-test/src/org/bandahealth/idempiere/base/test/modelevent/OrderModelEventTest.java [hunk 2]
```diff
@@ -52,4 +57,59 @@ public void orderIsCreatedCorrectlyWithModelEvents() throws Exception {
...
+
+		valueObject.getOrder().setDocAction(MOrder_BH.DOCACTION_Void);
+		assertTrue(valueObject.getOrder().processIt(MOrder_BH.DOCACTION_Void), "Order was successfully voided");
+
+		MInOut materialReceipt = new Query(valueObject.getContext(), MInOut.Table_Name, MInOut.COLUMNNAME_C_Order_ID +
+				"=?", valueObject.getTransactionName()).setParameters(valueObject.getOrder().get_ID()).first();
+
+		assertTrue(materialReceipt.getDocStatus().equalsIgnoreCase(MInOut.STATUS_Reversed), "Material receipt is voided");
+	}
 }
```
Comment (kwburnett) [inline idemp-base-test/src/org/bandahealth/idempiere/base/test/modelevent/OrderModelEventTest.java]:
```suggestion
}

```

File: idemp-base/src/org/bandahealth/idempiere/base/model/MPayment_BH.java [hunk 3]
```diff
@@ -348,4 +433,4 @@ public String getNHIF_Number() {
 	public void setNHIF_Number(String NHIF_Number) {
 		set_Value(COLUMNNAME_NHIF_Number, NHIF_Number);
 	}
-}
+}
```
Comment (kwburnett) [inline idemp-base/src/org/bandahealth/idempiere/base/model/MPayment_BH.java]:
```suggestion
}

```

**Review by andrewmoko — APPROVED** [review]

**Pull request merged**